src/services/fluence-aurora-connector/modules/fluence_aurora_connector/src/chain/log.rs

```diff
@@ -1,39 +1,106 @@
 use serde::{Deserialize, Serialize};
 
-use crate::chain::chain_data::ChainData;
+use crate::chain::chain_data::EventField::{Indexed, NotIndexed};
+use crate::chain::chain_data::LogParseError::{MissingToken, MissingTopic};
+use crate::chain::chain_data::{parse_chain_data, ChainData, LogParseError};
 use crate::chain::chain_event::ChainEvent;
 
-#[derive(Debug, Serialize, Deserialize)]
+#[derive(Debug, Serialize, Deserialize, Clone)]
 #[serde(rename_all = "camelCase")]
 pub struct Log {
     // Log arguments
     pub data: String,
     // The block number that contains this log
     pub block_number: String,
     // true when the log was removed, due to a chain reorganization. false if its a valid log.
+    #[serde(default)]
     pub removed: bool,
+    pub topics: Vec<String>,
 }
 
 pub fn parse_logs<U: ChainData, T: ChainEvent<U>>(logs: Vec<Log>) -> Vec<T> {
     logs.into_iter()
         .filter(|deal| !deal.removed)
-        .filter_map(|deal| parse_log::<U, T>(deal))
+        .filter_map(|deal| parse_log::<U, T>(deal).ok())
         .collect()
 }
 
-pub fn parse_log<U: ChainData, T: ChainEvent<U>>(log: Log) -> Option<T> {
+/// Parse Event Log to specified DTO
+///
+/// Logs consist of data fields, much like ADT. Fields can indexed and not indexed.
+///
+/// Data for indexed fields is encoded in 'log.topics', starting from 1th topic, i.e. 0th is skipped
+/// Data for non indexed fields is encoded in 'log.data'.
+///
+/// Indexed and non indexed data fields can be interleaved.
```

Comment:
Oh, now I understand
Suggest to duplicate these comments above EventField struct